storage/src/vespa/storage/distributor/externaloperationhandler.cpp

```diff
@@ -252,9 +253,10 @@ IMPL_MSG_COMMAND_H(ExternalOperationHandler, Put)
 
     auto handle = _mutationSequencer.try_acquire(cmd->getDocumentId());
     if (allowMutation(handle)) {
+        document::BucketSpace bucketSpace = cmd->getBucket().getBucketSpace();
         _op = std::make_shared<PutOperation>(*this,
-                                             _bucketSpaceRepo.get(cmd->getBucket().getBucketSpace()),
-                                             cmd, getMetrics().puts[cmd->getLoadType()], std::move(handle));
+                                             _bucketSpaceRepo.get(bucketSpace),
+                                             std::move(cmd), getMetrics().puts[loadType], std::move(handle));
```

Comment:
For all these callsites, `cmd` is a `const std::shared_ptr<T>&`, so the move won't have any effect in practice.

This is due to how the legacy storage links/chains work, where the message is always passed to handlers (as a const ref to the `shared_ptr`) and the handler returns whether it _handled_ the message after the fact. Since ownership is not known a-priori, moves cannot be done optimistically and using by-value would be too expensive.

Reply:
Yes, I figured that out. I just wanted to pull it as far out as possible, so when code is rewritten it will work optimal :)
At least one move into the PendingTrackerImpl takes effect.